qm-plugins/qm-vip/class-qm-vip-collector.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+class QM_VIP_Collector extends QM_Collector {
+
+	/**
+	 * @var string
+	 */
+	public $id = 'qm-vip';
+
+	/**
+	 * @return string
+	 */
+	public function name() {
+		return esc_html__( 'VIP', 'query-monitor' );
+	}
+
+	/**
+	 * @return void
+	 */
+	public function process() {
+		$this->process_version_file();
+	}
+
+	private function process_version_file() {
+		$version_file = WPMU_PLUGIN_DIR . '/.version';
+		if ( ! file_exists( $version_file ) ) {
+			$this->set_default_version_info();
+			return;
+		}
+		$version = file_get_contents( $version_file ); // phpcs:ignore WordPressVIPMinimum.Performance.FetchingRemoteData.FileGetContentsUnknown
+		if ( ! $version ) {
+			$this->set_default_version_info();
+			return;
+		}
+
+		$info          = json_decode( $version );
+		$branch        = $info->tag ?? null;
```

Comment:
On a separate note, here and forth we're expecting strings, so assigning `null` as a fallback value is not entirely semantically correct.